### PR TITLE
Feature/context menu chatpage

### PR DESCRIPTION
**Pull request by dmitry-zezik**
Not ready yet to merge.
1. Temp methods.

### CODE REVIEW

File: include/states/ChatState.h [hunk 1]
```diff
@@ -49,23 +49,33 @@ class ChatState : public QState
...
     void messageSent();
+    void isAdminChanged(const bool);
+    void isGroupChatChanged(const bool);
 
 private:
     void onMessageStatusChanged(const Message::Id &messageId, const Contact::Id &contactId, const Message::Status &status);
 
     Controllers *m_controllers;
     QString m_lastActivityText;
+    bool m_isAdmin;
```
Comment (fpohtmeh-github) [inline include/states/ChatState.h:77]: bool value isn't initalized after construction!

File: src/controllers/ChatsController.cpp [hunk 2]
```diff
@@ -139,6 +139,22 @@ void ChatsController::closeChat()
     setCurrentChat({});
 }
 
+void ChatsController::addParticipant(const UserId &userId)
+{
+    // addParticipant
```
Comment (fpohtmeh-github) [inline src/controllers/ChatsController.cpp:144]: Comment doesn't introduce new information, remove.

For such situation you can leave TODO comment:
`TODO: implement` or `TODO(d.zezik): implement`

File: src/qml/pages/ChatPage.qml [hunk 3]
```diff
@@ -75,6 +75,64 @@ Page {
                     color: Theme.secondaryTextColor
                 }
             }
+
+            ImageButton {
+                image: "More"
+                onClicked: {
+                    if (appState.isGroupChat) {
+                        if (appState.isAdmin) {
+                            groupChatAdminContextMenu.currentIndex = -1
```
Comment (fpohtmeh-github) [inline src/qml/pages/ChatPage.qml:84]: I guess context menu is opened with selected item and you reset selection.
Don't set currentIndex before every opening, move `currentIndex = -1` inside the ContextMenu component

File: src/qml/pages/ChatPage.qml [hunk 4]
```diff
@@ -75,6 +75,64 @@ Page {
...
+                    }
+                }
+
+                ContextMenu {
+                    id: stdChatContextMenu
+                    dropdown: true
+
+                    Action {
+                        text: qsTr("Delete chat")
+                        onTriggered: controllers.chats.addParticipant(/*userId*/)
```
Comment (fpohtmeh-github) [inline src/qml/pages/ChatPage.qml:102]: This shouldn't be commented, method has a parameter
Again, don't leave redundant comments

File: src/states/ChatState.cpp [hunk 5]
```diff
@@ -49,6 +49,9 @@ ChatState::ChatState(Controllers *controllers, VSQLastActivityManager *lastActiv
     connect(m_controllers->attachments(), &AttachmentsController::openPreviewRequested, this, &ChatState::requestPreview);
     connect(m_controllers->messages(), &MessagesController::messageStatusChanged, this, &ChatState::onMessageStatusChanged);
     connect(lastActivityManager, &VSQLastActivityManager::lastActivityTextChanged, this, &ChatState::setLastActivityText);
+//    TEMP SET
```
Comment (fpohtmeh-github) [inline src/states/ChatState.cpp:52]: test code goes to PR here, remove

File: src/states/ChatState.cpp [hunk 6]
```diff
@@ -65,6 +68,34 @@ void ChatState::setLastActivityText(const QString &text)
     emit lastActivityTextChanged(text);
 }
 
+bool ChatState::isAdmin() const
+{
+    return m_isAdmin;
+}
+
+void ChatState::setIsAdmin(const bool &adminValue)
```
Comment (fpohtmeh-github) [inline src/states/ChatState.cpp:76]: Optional fix. Be consistent in naming, setIsAdmin(const bool isAdmin)

Also don't use reference '&' for C++ primitive types

File: src/qml/pages/ChatPage.qml (hunk 3 above)
Comment (dmitry-zezik) [reply src/qml/pages/ChatPage.qml:84]: I copied you code from. Okay, I will fix it

File: include/states/ChatState.h (hunk 1 above)
Comment (dmitry-zezik) [reply include/states/ChatState.h:77]: Ok

File: src/controllers/ChatsController.cpp (hunk 2 above)
Comment (dmitry-zezik) [reply src/controllers/ChatsController.cpp:144]: +

File: src/qml/pages/ChatPage.qml (hunk 4 above)
Comment (dmitry-zezik) [reply src/qml/pages/ChatPage.qml:102]: +

File: src/states/ChatState.cpp (hunk 5 above)
Comment (dmitry-zezik) [reply src/states/ChatState.cpp:52]: +

File: src/states/ChatState.cpp (hunk 6 above)
Comment (dmitry-zezik) [reply src/states/ChatState.cpp:76]: +

**Review by fpohtmeh-github — APPROVED** [review]

**Pull request merged**